arch-install.bash

```diff
@@ -86,47 +90,106 @@ while true; do
+        dd if=/dev/zero of=/dev/mapper/to_be_wiped status=progress bs=1M
+        cryptsetup close to_be_wiped
+      ;;
+    esac
+  fi
 
-case "$bios" in
-  'uefi')
-    # efi part
-    sgdisk --new=1:0:+512M "$block_device"
-    # root
-    sgdisk --new=2:0:0 "$block_device"
-    ;;
-  'legacy')
-    # root
-    parted "$block_device" mklabel msdos mkpart primary 0% 100%
-    ;;
-esac
+  sgdisk --zap-all "$block_device"
 
-partitions=()
-query=$(sfdisk --json "$block_device")
-for k in $(jq '.partitiontable.partitions | keys | .[]' <<< "$query"); do
-  partitions+=("$(jq --argjson k "$k" -r '.partitiontable.partitions | .[$k] | .node' <<< "$query")")
-done
+  sgdisk --new=1:0:+512M "$block_device"
+  sgdisk --new=2:0:0 "$block_device"
 
-case "$bios" in
-  'uefi')
-    efi_fs="${partitions[0]}"
-    root_fs="${partitions[1]}"
-    umount --quiet "$efi_fs" "$root_fs" || true
-    mkfs.fat -F32 "$efi_fs"
-    mkfs.ext4 -F "$root_fs"
+  partitions=()
+  query=$(sfdisk --json "$block_device")
+  for k in $(jq '.partitiontable.partitions | keys | .[]' <<< "$query"); do
+    partitions+=("$(jq --argjson k "$k" -r '.partitiontable.partitions | .[$k] | .node' <<< "$query")")
+  done
+
+  echo "Setting up Encryption using Cryptsetup"
+
+  cryptsetup -y -v  luksFormat --type luks1 "${partitions[1]}"
+  cryptsetup open "${partitions[1]}" cryptroot
+  mkfs.ext4 /dev/mapper/cryptroot
+  mount /dev/mapper/cryptroot /mnt 
+  
+  case "$bios" in
+    'uefi')
+      mkfs.fat -F32 "${partitions[0]}"
     ;;
-  'legacy')
-    root_fs="${partitions[0]}"
-    umount --quiet "$root_fs" || true
-    mkfs.ext4 -F "$root_fs"
+    'legacy')
+      mkfs.ext4 "${partitions[0]}"
     ;;
-esac
+  esac
+  
+  mkdir /mnt/boot
+  mount "${partitions[0]}" /mnt/boot
 
-mount "$root_fs" /mnt
+else
+  sgdisk --zap-all "$block_device"
+
+  case "$bios" in
+    'uefi')
+      # efi part
+      sgdisk --new=1:0:+512M "$block_device"
+      # root
+      sgdisk --new=2:0:0 "$block_device"
+      ;;
+    'legacy')
+      # root
+      parted "$block_device" mklabel msdos mkpart primary 0% 100%
+      ;;
+  esac
+
+  partitions=()
+  query=$(sfdisk --json "$block_device")
+  for k in $(jq '.partitiontable.partitions | keys | .[]' <<< "$query"); do
+    partitions+=("$(jq --argjson k "$k" -r '.partitiontable.partitions | .[$k] | .node' <<< "$query")")
+  done
 
-if [[ $bios == 'uefi' ]]; then
-  mkdir -p /mnt/boot/efi
-  mount "$efi_fs" /mnt/boot/efi
+  case "$bios" in
+    'uefi')
+      efi_fs="${partitions[0]}"
+      root_fs="${partitions[1]}"
+      umount --quiet "$efi_fs" "$root_fs" || true
+      mkfs.fat -F32 "$efi_fs"
+      mkfs.ext4 -F "$root_fs"
+      ;;
+    'legacy')
+      root_fs="${partitions[0]}"
+      umount --quiet "$root_fs" || true
+      mkfs.ext4 -F "$root_fs"
+      ;;
+  esac
+
+
+  mount "$root_fs" /mnt
+
+  if [[ $bios == 'uefi' ]]; then
+    mkdir /mnt/boot
+    mount "$efi_fs" /mnt/boot
+    mkdir -p /mnt/boot/efi
```

Comment:
This is a leftover from when `/boot/efi` was being used for the separate EFI partition. Since `/boot` is now entirely its own partition, we can omit this line.

Reply:
You are correct. Will do